flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/batch/connectors/hive/HiveTableOutputFormat.java

```diff
@@ -198,6 +199,7 @@ private void readObject(ObjectInputStream in) throws IOException, ClassNotFoundE
 		tablePath = (ObjectPath) in.readObject();
 		partitionToWriter = new HashMap<>();
 		tableProperties = (Properties) in.readObject();
+		hiveVersion = jobConf.get(HiveCatalogValidator.CATALOG_HIVE_VERSION, HiveShimLoader.getHiveVersion());
```

Comment:
I'm curious whether open() will be called after deserialization. If so, this isn't not necessary. Secondly, having this here makes deserialization not symmetric to serialization. If we have to have this, I'd include hiveVersion in the serialized fields.

Reply:
`open` is called on TM but not on JM, where `finalizeGlobal` is performed. That's why I initialize hiveVersion in `readObject` and constructor (to be safe).
I'll serialize this field instead of getting it from jobConf.